### PR TITLE
fix: add support for initializing this public struct

### PR DESCRIPTION
**Pull request by raulriera**
## Description

At the moment it seems to not be possible to use this struct from outside of the library, even tho its marked as public

```shell
'DataSlice' initializer is inaccessible due to 'internal' protection level
```

### CODE REVIEW

**Review by TrGiLong — APPROVED** [review]

**Review by bigearsenal — APPROVED** [review]

**Comment by raulriera** [comment]
Thanks for the reviews everyone! but I have no write access so I can't merge this on my own

**Pull request merged**
